### ISSUE TITLE
Resolve the conflicting responsibilities between api/controller and module of Node 

### ISSUE THREAD

**Issue by MaciejBaj** [LiskHQ/lisk#2783]
### Expected behavior
There should be no dependency between `api/controllers/node`and `modules/node`.
Detach the dependencies by moving all of the API logic from `modules/node` to api/controllers/node`. 

As the result, the `modules/node` should be removed, as it's not required by any other module.

All of the node module dependencies needs to be replaced by the direct calls to chain module when implemented. Until chain module is not implemented as an independent module, the `ToDo` placeholders should be left when required. After all, the depencencies should be replaced by modules/component/entities as explained in the comments:
```
Node.prototype.onBind = function(scope) {
	modules = {
		blocks: scope.blocks, //should be replaced by storage.entity.block
		loader: scope.loader, //should be replaced by System Component (to verify)
		peers: scope.peers, //should be replaced by P2P Component/Module
		system: scope.system, //should be replaced by System Component
		delegates: scope.delegates, //should be replaced by Chain Module
	};
	loaded = true;
};
```

### Actual behavior
There is a conflict of responsibilities between the logic realised by `modules/node` and `api/controllers/node`which requires the dependency to the module while executing:

- `NodeController.getConstants`
- `NodeController.getStatus`
- `NodeController.getStatus`
- `NodeController.getForgingStatus`
- `NodeController.updateForgingStatus`

Inside each of this function there is a call to `modules.node.shared` property.

### Which version(s) does this affect? (Environment, OS, etc...)
`*`

**Comment by MaciejBaj** [comment]
Merged into 1.5.0.
